app/Http/Controllers/JobStatusController.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Exceptions\StateMachineException;
+use App\Models\JobPoster;
+use App\Models\JobPosterStatusHistory;
+use App\Models\Lookup\JobPosterStatus;
+use App\Services\JobStatusTransitions;
+use Illuminate\Http\Request;
+
+class JobStatusController extends Controller
+{
+    protected function transitionJobStatus(Request $request, JobPoster $job, string $to)
+    {
+        $transitions = new JobStatusTransitions();
+
+        $user = $request->user();
+        $fromStatus = $job->job_poster_status;
+        $from = $fromStatus->name;
+
+        // Ensure state transition is legal.
+        if (!$transitions->canTransition($user, $from, $to)) {
+            throw new StateMachineException('Illegal state transition');
+        }
+
+        // Save new status on job.
+        $toStatus = JobPosterStatus::where('name', $to)->first();
+        $job->job_poster_status_id = $toStatus->id;
+        $job->save();
+
+        // Save transition history.
+        $transition = new JobPosterStatusHistory();
+        $transition->job_poster_id = $job->id;
+        $transition->user_id = $user->id;
+        $transition->from_job_poster_status_id = $fromStatus->id;
+        $transition->to_job_poster_status_id = $toStatus->id;
+        $transition->save();
+
+        return $request->ajax()
+            ? response()->json(['status' => 'ok'])
+            : back();
+    }
+
+    public function setJobStatus(Request $request, JobPoster $jobPoster, string $status)
```

Comment:
I separated it because I wanted to allow flexibility in how the entry point gets the destination status. I wasn't sure if the url should specify the status to transition to (the current solution) or whether that should be part of a payload (ie {status: 'completed'}). Do you have any preference?

Reply:
I think payload probably makes more sense? That way we don't have to maintain `n` routes and changes to the statuses can be more seamless.

Reply:
If its only about maintaining routes, the current setup is pretty good:
```
Route::post(
        'jobs/{jobPoster}/set-status/{status}',
        'JobStatusController@setJobStatus'
    )
```